### DISCUSSION TITLE
GOS and GND in MTEX

### DISCUSSION THREAD

**Discussion by wuwuhd** [mtex-toolbox/mtex#1989] [Ask Anything]
Dear Ralf,

I am sorry to trouble you. I am calculating GOS and GND denisty in a dual phase (mixtures of fcc asutenite and bcc martensite) steel, but I only know how to obtain GOS and GNDs in a single phase material. Could you give a coding exmaple to help me solve this problem?

Thank you very much in davance!

Best whishes to you!

wu

this is my code.

`[grains,ebsd('Iron bcc').grainId] = calcGrains(ebsd(('Iron bcc')));

ebsd(grains(grains.grainSize<=5)) = [];

[grains,ebsd('Iron bcc').grainId] = calcGrains(ebsd('Iron bcc'),'threshold',[1*degree,10*degree]);

grains = smooth(grains,5);

grod=ebsd.calcGROD(grains)

plot(ebsd,grod.angle./degree,'micronbar','off')
mtexColorbar('title','misorientation angle to meanorientation in degree')
mtexColorMap LaboTeX

hold on
plot(grains.boundary,'lineWidth',1.5)
plot(grains.innerBoundary,'edgeAlpha',grains.innerBoundary.misorientation.angle / (5*degree))
hold off
GOS = grainMean(ebsd, grod.angle);

plot(grains, GOS ./ degree)
mtexColorbar('title','GOS in degree')
`

**Comment by kilir** [comment]
Hi,
there's `grains.GOS` which operates on all phases. However in the case of GND, you simply compute for each phase you have
e.g.
```
% 1st phase
ebsd_f = ebsd('f').gridify;
kappa_f = ebsd_f.curvature
%... and so on and so on...

% 2nd phase
ebsd_e = ebsd('e').gridify;
kappa_e = ebsd_e.curvature

%... and so on and so on...
```
Hope that helps.

Cheers,
Rüdiger

**Reply by wuwuhd** [reply]
Thanks, Rüdiger
I'm sorry that I am a beginner of MTEX. The grains.GOS mean...? Can you be more specific?
Thanks for your help sincerely,

wu

**Reply by kilir** [reply]
It's a property computed during grain reconstruction. However, you're approach also works, simply compute grains for all phases and then use `ebsd(xy)` and `grains(xy)`.
Cheers,
Rüdiger

**Reply by wuwuhd** [reply]
OK, I am sorry, but can you tell me where I can find the grains.GOS  and use it? And how do I use ebsd(xy) and grains(xy)? It is a example?
sincerely,
wu

**Reply by kilir** [reply]
Hi,

see here: https://mtex-toolbox.github.io/GrainOrientationParameters.html
When you compute grains, the GOS is computed as a property and available as the property `grains.GOS`.

I used xy as a placeholder for whatever phase names would be of interest. So for example `ebsd('rutile')` would just be the ebsd of the phase Rutile and `grains('Manganotantalite')` would just be adressing the grains of the phase Manganotantalite.

Cheers,
Rüdiger
